src/FirmwarePlugin/FirmwarePluginManager.cc

```diff
@@ -42,11 +42,46 @@ FirmwarePluginManager::~FirmwarePluginManager()
 
 }
 
-FirmwarePlugin* FirmwarePluginManager::firmwarePluginForAutopilot(MAV_AUTOPILOT autopilotType)
+FirmwarePlugin* FirmwarePluginManager::firmwarePluginForAutopilot(MAV_AUTOPILOT autopilotType, MAV_TYPE vehicleType)
 {
     switch (autopilotType) {
         case MAV_AUTOPILOT_ARDUPILOTMEGA:
-            return APMFirmwarePlugin::instance();
+            switch (vehicleType) {
+                case MAV_TYPE_QUADROTOR:
+                case MAV_TYPE_HEXAROTOR:
+                case MAV_TYPE_OCTOROTOR:
+                case MAV_TYPE_TRICOPTER:
+                    return ArduCopterFirmwarePlugin::instance();
+                    break;
+                // FIXME: The remainder of these need to be correctly assigned and new plugin classes created as needed.
```

Comment:
I don't know whether I understood it correctly. Looks like you want me to
have a baseAPM plugin and special implementations for MAV_TYPES. Is that so?

On Sat, Oct 10, 2015 at 10:19 PM, Don Gagne <notifications@github.com>
wrote:

> In src/FirmwarePlugin/FirmwarePluginManager.cc
> <https://github.com/mavlink/qgroundcontrol/pull/1940#discussion_r41698946>
> :
>
> >  {
> >      switch (autopilotType) {
> >          case MAV_AUTOPILOT_ARDUPILOTMEGA:
> > -            return APMFirmwarePlugin::instance();
> > +            switch (vehicleType) {
> > +                case MAV_TYPE_QUADROTOR:
> > +                case MAV_TYPE_HEXAROTOR:
> > +                case MAV_TYPE_OCTOROTOR:
> > +                case MAV_TYPE_TRICOPTER:
> > +                    return ArduCopterFirmwarePlugin::instance();
> > +                    break;
> > +                // FIXME: The remainder of these need to be correctly assigned and new plugin classes created as needed.
>
> This needs to be completed by assigning the correct set to ArduCopter (I
> may have missed some). And also creating the correct sets for the other
> firmware types.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mavlink/qgroundcontrol/pull/1940/files#r41698946>.
>



-- 
Regards,
pritam